improver/generate_ancillaries/generate_derived_solar_fields.py

```diff
@@ -429,6 +429,17 @@ def _create_solar_radiation_cube(
         )
 
         attrs = generate_mandatory_attributes([target_grid])
+        attrs["source"] = "IMPROVER"
+        # If title in source grid, this will transfer through generate_mandatory_attributes.
+        # Here we assume that anything following "on" in title attribute describes the grid,
+        # which we can pass onto the clearsky solar radiation cube.
+        target_grid_title = attrs.get("title", "")
+        grid_in_title = target_grid_title.rsplit("on ", maxsplit=1)
+        if len(grid_in_title) > 1:
+            grid_details = f" on {grid_in_title[-1]}"
+        else:
+            grid_details = ""
```

Comment:
Hi @benowen-bom. I think it would be better to do something like we do in the `spot-extract` CLI: https://github.com/benowen-bom/improver/blob/c18253d5bc52ae4bb20d6f0eee952016e0bfa8a8/improver/cli/spot_extract.py#L95
There, a keyword argument can be specified which is passed through to the title attribute of the output cube.

Your pattern match above might be better if the pattern had a leading space: `" on "` to prevent matching a partial word.

Reply:
Thanks @MoseleyS, your proposed option will avoid the issue of unexpected behaviour that may arise from splitting up the `"title"` sting from the target grid. So I think I will go with that. I'll add in an additional acceptance test to check the behaviour in this case as this change will flow up to the CLI interface.

I also realised I forgot to add the cube names of my new ancillaries to this list (as requested previously), so I'll do that too. With these changes, hopefully we're basically there.